### PR TITLE
adding `find_statistic` function

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
An initial draft of the function; need more work

To do-
- [x] Adding tests
- [x] Checking edge cases (e.g. `gamm4`)

### CODE REVIEW

File: R/find_statistic.R [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+#' m <- lm(mpg ~ wt + cyl + vs, data = mtcars)
+#'
+#' find_statistic(m)
+#' @export
+find_statistic <- function(x, ...) {
+
+  # model object check --------------------------------------------------------
+
+  # check if the object is a model object; if so, quit early
+  if (isFALSE(is_model(x))) {
```
Comment (strengejacke) [inline R/find_statistic.R]: I would not use `isFALSE()` because it was implemented rather late in R and we want to be more backwards compatible than just R 3.5 or 3.6.

File: R/find_statistic.R [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  # statistic check -----------------------------------------------------------
+
+  if (class(x)[[1]] %in% unsupported.mods) return(NULL)
+  if (class(x)[[1]] %in% t.mods) return("t-statistic")
+  if (class(x)[[1]] %in% z.mods) return("z-statistic")
+  if (class(x)[[1]] %in% f.mods) return("F-statistic")
+  if (class(x)[[1]] %in% chi.mods) return("chi-squared statistic")
+
+  if (class(x)[[1]] %in% g.mods) {
+    if (class(x)[[1]] == "glm" && summary(x)$family$family[[1]] %in% g.t.mods) {
```
Comment (strengejacke) [inline R/find_statistic.R]: You could use `model_info()` instead of `summary()`, which might be more stable.

**Comment by strengejacke** [comment]
> ── 1. Failure: find_statistic (@test-gam.R#112)  ───────────────────────────────
find_statistic(m1) not identical to "F-statistic".
1/1 mismatches
x[1]: "t-statistic"
y[1]: "F-statistic"

I think this is easy to address.

> ── 2. Error: find_statistic (@test-gamm4.R#120)  ───────────────────────────────

here the problems seems that gamm4 are _lists_, containing two models (gam and lme). We may add a helper-method that checks if model is a gam or gamm4 model, see e.g. here: https://github.com/strengejacke/ggeffects/blob/7624a2dbcdb625cc688991fce4d1050707863bd8/R/utils.R#L276

> ── 3. Error: find_statistic (@test-GLMMadaptive.R#311)  ────────────────────────

Not sure, must look closer at the code.

**Comment by IndrajeetPatil** [comment]
Travis tests are doing well.

One test is failing on Windows builds though:
https://ci.appveyor.com/project/IndrajeetPatil/insight/build/job/wfd4xr9xw72li23f

image

**Comment by IndrajeetPatil** [comment]
@strengejacke, @DominiqueMakowski Can someone have a look at this? 
It has been pending for a while now and I want to avoid any further merge conflicts.

**Comment by IndrajeetPatil** [comment]
Thanks. Yeah, this can be merged. 

As for linter, we can even just remove the `lintr` file later if we feel it is too strict and is cluttering the `PR` comments section. 

I just look at it as a good way to see some recommendations about good coding practices. In my workflow, I have definitely found a few instances where it has saved me from making a mistake.

**Comment by pdwaggoner** [comment]
Sorry for my delay - this looks good to me. Just reviewed the changes. Great work

**Comment by strengejacke** [comment]
I typicall wait at least ~ 2 months, so it would be by the end of September.

**Comment by strengejacke** [comment]
Yes, we can also aim at releasing by mid of September. There are no pending issues, it's more that I try to not submit too often, as stated by the CRAN team. But if it would help you, I can submit around ~15th September. Any specific date you prefer?
